src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,454 @@
+          if (pointerValue === currentValues[activeThumbIndex]) {
+            return;
+          }
+          const newValues = [...currentValues];
+          newValues[activeThumbIndex] = pointerValue;
+          setCurrentValues(newValues);
+          callback?.(newValues);
+        }
+      },
+      [activeThumbIndex, min, max, step, getAllowableThumbRange, currentValues],
+    );
+
+    const handlePointerMove = React.useCallback(
+      (event: PointerEvent): void => {
+        updateThumbValue(event, onUpdate);
+      },
+      [onUpdate, updateThumbValue],
+    );
+
+    // function called by Thumb keyboard processing
+    const onThumbValueChanged = React.useCallback(
+      (index: number, value: number) => {
+        if (currentValues[index] === value) {
+          return;
+        }
+        const newValues = [...currentValues];
+        newValues[index] = value;
+        setCurrentValues(newValues);
+        onChange?.(newValues);
+      },
+      [currentValues, onChange],
+    );
+
+    const onThumbActivated = React.useCallback((index: number) => {
+      setActiveThumbIndex(index);
+    }, []);
+
+    const handlePointerUp = React.useCallback(
+      (event: PointerEvent) => {
+        updateThumbValue(event, onChange);
+        setActiveThumbIndex(undefined);
+        event.preventDefault();
+        event.stopPropagation();
+      },
+      [onChange, updateThumbValue],
+    );
+
+    const handlePointerDownOnSlider = React.useCallback(
+      (event: React.PointerEvent) => {
+        if (containerRef.current) {
+          const percent = getPercentageOfRectangle(
+            containerRef.current.getBoundingClientRect(),
+            event.pageX,
+          );
+          let pointerValue = min + (max - min) * percent;
+          pointerValue = roundValueToClosestStep(pointerValue, step, min);
+
+          const closestValueIndex = getClosestValueIndex(
+            currentValues,
+            pointerValue,
+          );
+          const [minVal, maxVal] = getAllowableThumbRange(closestValueIndex);
+          pointerValue = getBoundedValue(pointerValue, minVal, maxVal);
+          if (pointerValue === currentValues[closestValueIndex]) {
+            return;
+          }
+          const newValues = [...currentValues];
+          newValues[closestValueIndex] = pointerValue;
+          setCurrentValues(newValues);
+          onChange?.(newValues);
+        }
+      },
+      [min, max, step, currentValues, getAllowableThumbRange, onChange],
+    );
+
+    const pointerMoveFunctionRef = React.useRef<
+      (event: PointerEvent) => void
+    >();
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      if (pointerMoveFunctionRef.current && ownerDoc) {
+        ownerDoc.removeEventListener(
+          'pointermove',
+          pointerMoveFunctionRef.current,
+        );
+      }
+      pointerMoveFunctionRef.current = handlePointerMove;
```

Comment:
Yeah..how sure you are this always works as expected? :D
Cause from what I know, useEffect code is ran asynchronously and you cannot predict the order.. So theoretically, what could happen that firstly the addition of the event listener is ran and then the removal part.. Havent you noticed if sometimes stuff brakes??
I would have expected to see the removal of the event listeners in the same `useEffect` hook as the addition..